stacks/psa-stack.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Temporary stack for the old Pinning Service API (PSA) that maps root CIDs
+ * to CAR files the complete DAGs are stored in.
+ */
+import { Function } from 'sst/constructs'
+
+/** @param {import('sst/constructs').StackContext} context */
+export function PSAStack ({ stack }) {
+  stack.setDefaultFunctionProps({
+    runtime: 'nodejs20.x',
+    architecture: 'arm_64'
+  })
+
+  const hashFunction = new Function(stack, 'hash', {
+    handler: 'shardutil/functions/hash.handler',
+    url: { cors: true, authorizer: 'none' },
+    memorySize: '4 GB',
+    timeout: '15 minutes'
+  })
+
+  hashFunction.attachPermissions(['s3:GetObject'])
+
+  const downloadFunction = new Function(stack, 'download', {
+    handler: 'shardutil/functions/download.handler',
+    url: { cors: true, authorizer: 'none' }
+  })
+
+  downloadFunction.attachPermissions(['s3:GetObject'])
+
+  stack.addOutputs({
+    hashFunctionURL: hashFunction.url,
```

Comment:
do these need a domain name?

Reply:
Yes, good shout - otherwise they will change when we deploy.

Reply:
Going to have to setup a redirect for this manually since we want to allow execution of >30s (hashing big files). The AWS API gateway has a limit of 30s, which I think is now configurable, but I don't think our SST version supports it yet.